scipy/signal/signaltools.py

```diff
@@ -501,17 +621,39 @@ def convolve(in1, in2, mode='full'):
         # Convolution is commutative; order doesn't have any effect on output
         volume, kernel = kernel, volume
 
+    if method == 'fft' and not _numeric_arrays([volume, kernel]):
```

Comment:
so I think this line is not necessary?  if user requests fft and fftconvolve can't handle the inputs, let it raise the exception itself.  currently it just converts things to float if it can:

```
a = array([Decimal('0.1'), Decimal('0.2')])

b = [4]

fftconvolve(a, b)
Out[83]: array([ 0.40000000000000008,  0.80000000000000004])

convolve(a, b, method='fft')
Traceback (most recent call last):
...
ValueError: convolve can only be called with method=="fft" when arrays consist of numeric elements (i.e., int/float/etc)
```